dolomite_engine/hf_models/modeling_utils/position_embedding/rope.py

```diff
@@ -24,10 +24,10 @@ def __init__(
 
     def forward(self, seq_len: int, dtype: torch.dtype, device: torch.device) -> tuple[torch.Tensor, torch.Tensor]:
         if seq_len > self.max_seq_len_cached:
-            self._set_cos_sin_cache(seq_len=seq_len, device=device, dtype=dtype)
+            self._set_cos_sin_cache(seq_len=seq_len, dtype=dtype)
 
-        cos = self.cos_cached[:seq_len].to(dtype)
-        sin = self.sin_cached[:seq_len].to(dtype)
+        cos = self.cos_cached[:seq_len].to(dtype).to(device)
+        sin = self.sin_cached[:seq_len].to(dtype).to(device)
```

Comment:
@yikangshen is this change needed?
or do we just need to remove device=device from above line?

Reply:
Remove device=device will create some other issue while loading checkpoints for inference.